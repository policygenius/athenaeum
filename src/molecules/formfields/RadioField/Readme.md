RadioField Example:

```jsx
    class RadioFieldExample extends React.Component {
      constructor(props) {
        super(props);

        this.state = {
          value: '',
        };

        this.onChange = this.onChange.bind(this);
      }

      onChange(value) {
        this.setState({
          value
        });
      }

      render() {
        return (
          <div>
            <RadioField
              label='Fire extinguisher'
              input={{
                name: 'radio_test',
                value: this.state.value,
                onChange: this.onChange
              }}
              radioValue='fire_ext'
            />

            <Spacer size={18} />

            <RadioField
              label='Flame thrower'
              input={{
                name: 'radio_test',
                value: this.state.value,
                onChange: this.onChange
              }}
              radioValue='flame_throw'
            />
          </div>
        )
      }
    }

    <RadioFieldExample />
```
