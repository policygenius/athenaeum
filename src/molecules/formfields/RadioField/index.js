import React from 'react';
import PropTypes from 'prop-types';
import camelCase from 'lodash/camelCase';
import omit from 'lodash/omit';
import classnames from 'classnames';

import styles from './radio_field.module.scss';

function RadioField(props) {
  const {
    label,
    className,
    input,
    radioValue
  } = props;

  const classes = [
    styles['radio-field'],
    className,
  ];

  const labelClasses = [
    styles['label'],
    input.value === radioValue && styles['checked'],
    input.value && input.value !== radioValue && styles['not-selected'],
  ];

  return (
    <label
      className={classnames(...classes)}
      htmlFor={`radio-${camelCase(radioValue)}`}
    >
      <input
        type='radio'
        name={input.name}
        id={`radio-${camelCase(radioValue)}`}
        className={classnames(styles['radio'])}
        onChange={() => input.onChange(radioValue)}
        {...omit(input, 'onChange')}
      />
      <span className={classnames(...labelClasses)}>{ label }</span>
    </label>
  );
}


RadioField.propTypes = {
  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,

  /**
   * label for field directly next to the button
   */
  label: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ]),
  /**
   * The props under the meta key are metadata about the state of this field that `redux-form` tracks.
   */
  meta: PropTypes.object,
  /**
   * The props under the input key are passed from `redux-form` and spread into `<input />`.
   */
  input: PropTypes.object.isRequired,

  /**
   * Value for radio field
   */
  radioValue: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]).isRequired
};

export default RadioField;
