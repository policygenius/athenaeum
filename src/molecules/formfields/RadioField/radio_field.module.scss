$size: ru(.75);
$gap: ru(.5);
$checked-size: ru(.5);

.radio-field {
  @include typography-b-8;

  margin: 0;
  display: block;
  position: relative;
  min-height: $size;
  padding-left: $size + $gap;
  user-select: none;

  &:before {
    position: absolute;
    top: (ru(1) - $size) / 2;
    left: 0;
    width: $size;
    height: $size;
    overflow: hidden;
    border: 1px solid color('neutral-3');
    background-color: color('primary-2');
    border-radius: 50%;
    content: '';
  }
}

.label {
  display: inline-block;
  color: color('primary-3');
  cursor: pointer;

  &:after {
    content: '';
    opacity: 0;
    position: absolute;
    top: (ru(1) - $size) / 2 + ($size - $checked-size) / 2;
    left: ($size - $checked-size) / 2;
    width: $checked-size;
    height: $checked-size;
    border-radius: 50%;
    background-color: color('secondary-2');
  }

  &.checked {
    &:after { opacity: 1; }
  }

  &.not-selected {
    color: color('neutral-3');
  }
}

.radio {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  // width: $size + $gap;
  height: $size;
  opacity: 0;
  margin: 0;
  cursor: pointer;
  z-index: 10;
  pointer-events: all;
}

.radio:disabled {
  + .label {
    cursor: not-allowed;
    color: color('neutral-3');
  }
}
