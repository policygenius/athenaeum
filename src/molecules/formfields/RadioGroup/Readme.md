### RadioGroup Example:

```jsx
    class RadioGroupExample extends React.Component {
      constructor(props) {
        super(props);

        this.state = {
          value: '',
        };

        this.onChange = this.onChange.bind(this);
      }

      onChange(value) {
        this.setState({
          value
        });
      }

      render() {
        return (
          <RadioGroup
            label='Test label'
            subLabel='Sublabel action'
            meta={{}}
            input={{}}
            tooltip='Hello friend'
          >
            <RadioField
              label='Burgers'
              radioValue='burgers'
              input={{
                name: 'food',
                value: this.state.value,
                onChange: this.onChange,
              }}
            />
            <RadioField
              label='Hot Dogs'
              radioValue='dogs'
              input={{
                name: 'food',
                value: this.state.value,
                onChange: this.onChange,
              }}
            />
          </RadioGroup>
        )
      }
    }

    <RadioGroupExample />
```
