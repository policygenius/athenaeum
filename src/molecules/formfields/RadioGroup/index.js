import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import ErrorMessage from 'atoms/ErrorMessage';
import Text from 'atoms/Text';
import { renderTooltip } from 'utils/fieldUtils';

import styles from './radio_group.module.scss';

function RadioGroup(props) {
  const {
    children,
    className,
    label,
    subLabel,
    input,
    meta,
    tooltip,
  } = props;

  const showErrorMessage = (meta.visited && !meta.active) || meta.submitFailed;
  const message = meta && showErrorMessage && (meta.error || meta.warning);

  const classes = [
    styles['radio-group'],
    meta && meta.active && styles['focused'],
    meta && showErrorMessage && meta.error && !meta.active && styles['hasError'],
    className,
  ];

  return (
    <div>
      <div
        className={classnames(...classes)}
        onBlur={(e) => {
          if (!e.relatedTarget) input.onBlur();
        }}
        onFocus={input && input.onFocus}
      >
        <div className={styles['label-wrapper']}>
          <div className={styles['label']}>
            <span>{label}</span>
            { tooltip && renderTooltip(tooltip, styles['tooltip'], styles['tooltip-icon']) }
          </div>

          {
            subLabel &&
              <Text
                size={10}
                font='b'
              >
                {subLabel}
              </Text>
          }
        </div>
        {
          React.Children.map(children, child =>
            <div className={styles['field']}>
              { child }
            </div>
          )
        }
      </div>

      <ErrorMessage
        condition={!!message}
        message={message}
      />
    </div>
  );
}


RadioGroup.propTypes = {
  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,

  /**
   * Label for group
   */
  label: PropTypes.string,

  /**
   * label to appear directly under the label
   */
  subLabel: PropTypes.string,

  /**
   * Input props provided by ReduxForm
   */
  input: PropTypes.object,

  /**
   * meta props provided by ReduxForm
   */
  meta: PropTypes.object,

  /**
   * either a handler for clicking the tooltip, or text to go in the tooltip for the label
   */
  tooltip: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.string,
  ]),
};

export default RadioGroup;
