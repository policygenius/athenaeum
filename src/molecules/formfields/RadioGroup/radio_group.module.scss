@import '../shared/formfields.module.scss';

.radio-group {
  @extend %form-field;
}

.label-wrapper {
  padding: ru(.5) ru(.75);

  border-bottom: 1px solid color('neutral-4');
}

.label {
  @include typography-b-10(true);

  display: flex;
}

.field {
  border-bottom: 1px solid color('neutral-4');
  padding: ru(.5) ru(.75);

  &:last-child {
    border: 0;
  }
}

.tooltip {
  margin-left: auto;
}

.tooltip-icon {
  @extend %tooltip-icon;
}

/* --  state:focus  -- */
.focused {
  border: 1px solid color('secondary-2');
}

/* --  state:error  -- */
.hasError {
  border: 1px solid color('primary-1');
}

@media #{$tablet} {
  .label {
    @include typography-b-8(true);
  }
}
