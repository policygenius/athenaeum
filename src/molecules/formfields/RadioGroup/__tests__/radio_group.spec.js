/* eslint-disable no-extra-parens */
import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import RadioGroup from 'molecules/formfields/RadioGroup';
import RadioField from 'molecules/formfields/RadioField';

describe('<RadioGroup />', () => {
  let component;

  beforeEach(() => {
    component =
      (<RadioGroup
        label='Test label'
        subLabel='Sublabel action'
        meta={{}}
        input={{}}
        tooltip='Hello friend'
      >
        <RadioField
          label='Burgers'
          radioValue='burgers'
          input={{
            name: 'food',
            value: null,
            onChange: () => {},
          }}
        />
        <RadioField
          label='Hot Dogs'
          radioValue='dogs'
          input={{
            name: 'food',
            value: null,
            onChange: () => {},
          }}
        />
      </RadioGroup>)
    ;
  });

  it('renders correctly', () => {
    const wrapper = mount(component);

    expect(wrapper.type()).toEqual(RadioGroup);
  });

  it('renders correctly', () => {
    const actual = renderer.create(component).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
