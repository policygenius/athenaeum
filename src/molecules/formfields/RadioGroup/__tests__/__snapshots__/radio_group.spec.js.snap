// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup /> renders correctly 1`] = `
<div>
  <div
    className="radio-group"
    onBlur={[Function]}
  >
    <div
      className="label-wrapper"
    >
      <div
        className="label"
      >
        <label>
          Test label
        </label>
        <span>
          <span
            className="tooltip-wrapper tooltip"
            onClick={[Function]}
          >
            <span
              className="icon-wrapper tooltip-icon"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            >
              <img
                alt="tooltip"
                className="img"
                src="https://static.policygenius.com/svgs/tooltip.svgz"
                title="tooltip"
              />
            </span>
            <span
              className="hover-message right"
            >
              Hello friend
            </span>
          </span>
        </span>
      </div>
      <p
        className="primary-3 type-b-10-medium inherit"
      >
        Sublabel action
      </p>
    </div>
    <div
      className="field"
    >
      <label
        className="radio-field"
        htmlFor="radio-burgers"
      >
        <input
          className="radio"
          id="radio-burgers"
          name="food"
          onChange={[Function]}
          type="radio"
          value="dogs"
        />
        <span
          className="label not-selected"
        >
          Burgers
        </span>
      </label>
    </div>
    <div
      className="field"
    >
      <label
        className="radio-field"
        htmlFor="radio-dogs"
      >
        <input
          className="radio"
          id="radio-dogs"
          name="food"
          onChange={[Function]}
          type="radio"
          value="dogs"
        />
        <span
          className="label checked"
        >
          Hot Dogs
        </span>
      </label>
    </div>
  </div>
  <span />
</div>
`;
